client.go

```diff
@@ -27,26 +40,42 @@ func Post(reqUrl string, auth string, body interface{}, contentType string) (res
 
 	if err != nil {
 		log.Fatalln(err)
+
 	}
 
+	//b, err := httputil.DumpRequest(req, true)
+	//
 	//fmt.Printf("Request = %s\n", string(b))
 
 	return handleResp(req)
 }
 
-func Put(reqUrl string, auth string, body interface{}, contentType string) (respBodyObj ResponseBody, err error) {
+func Put(reqUrl string, auth string, body interface{}, contentType string, bufferBody *bytes.Buffer) (respBodyObj ResponseBody, err error) {
 	postBody, _ := json.Marshal(body)
 	requestBody := bytes.NewBuffer(postBody)
+	var req *http.Request
 	log.WithFields(log.Fields{
 		"body": string(postBody),
 	}).Debug("The request body")
-	req, err := http.NewRequest("PUT", reqUrl, requestBody)
+	if bufferBody != nil {
```

Comment:
Same as above, This if else can be simplified as follows:
```
requestBody := bytes.NewBuffer(postBody)
if bufferBody != nil {
  requestBody = bufferBody
} 
request, reqError := http.NewRequest("POST", reqUrl,  requestBody)
```
The request per-se doesn't change only one param is different.